aarch64/root/firmware-update.sh

```diff
@@ -54,13 +84,13 @@ echo "Firmware File: $fileName"
 echo " "
 echo "Are the above device and firmware values correct?"
 read -p "Enter Y to proceed, any other entry to exit: " correctVal
+echo " "
 
-if [ "$correctVal" = "Y" ] || [ "$correctVal" = "y" ]; then
-        echo " "
+if [[ $correctVal == [yY] ]]; then
         echo "Flashing..."
         echo " "
-        /usr/bin/GCFFlasher_internal -d $deviceName -f /usr/share/deCONZ/firmware/$fileName
-        
+        $FLASHER -d $deviceName -f "$filePath"
+
         retVal=$?
         if [ $retVal != 0 ]; then
```

Comment:
Are () or [] correct here to check retVal?

Reply:
As far as I know:
1. Single `[ expr ]` will call the `test()` function
2. Double `[[ expr ]]` is a bash-internal.
3. Double `(( expr ))` is an arithmetic expression - `(( 9 > 10 ))` will evaluate differently than `[[ 9 > 10 ]]`. The latter checks for lexicographic order ...

As an addidtion: In arithmetic expression you don't need to use the `$` for numeric variables.

So as short answer: both work and are correct. The difference may be in comparing strings or numbers ...

Reply:
Sounds correct to me. Thanks!